admin/workspace-providers/deployment.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Deployment
+description: Learn how to deploy a workspace provider.
+state: beta
+---
+
+This article walks you through the process of deploying a workspace provider to
+a [Kubernetes cluster](../../setup/kubernetes/index.md).
+
+[Workspace Providers](index.md) are logical groups of resources to which
+developers can deploy environments. Like the Coder deployment, workspace providers
+are deployed via a helm chart to the Kubernetes cluster where you'd like to
+provision new environments.
+
+## Dependencies
+
+Install the following dependencies if you haven't already:
+
+- [Coder CLI](../../cli/installation.md)
+- [helm](https://helm.sh/docs/intro/install/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Requirements
+
+1. Workspace providers **must have a hostname set** that is a subdomain of the
+   Coder deployment. For example, if the Coder deployment's hostname is
+   `coder.example.com`, the workspace provider's hostname must match the format
+   `*.coder.example.com`.
+1. The main Coder deployment and the workspace provider must be able to
+   communicate bi-directionally via their respective hostnames.
+1. The workspace provider HTTP protocol must match the protocol of the Coder
+   deployment Access URL.
+1. The Kubernetes cluster address must be reachable from the Coder deployment.
+
+## Connecting to the Cluster
+
+To add a Kubernetes cluster as a workspace provider, you must first make sure
+that you're connected to the cluster you want to expand into. Run the following
+command:
+
+```bash
+kubectl config current-context
+```
+
+Confirm that your current kubectl context correct before continuing; otherwise,
+connect to the correct context.
+
+## Creating the Coder Namespace (Optional)
+
+We recommend running workspace providers in a separate
+[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/);
+to do so, run:
+
+```bash
+kubectl create namespace coder
+```
+
+Next, change the kubectl context to point to your newly created namespace:
+
+```bash
+kubectl config set-context --current --namespace=coder
+```
+
+## Creating the New Workspace Provider
+
+Using the Coder CLI, create a new workspace provider in the `pending` state.
+
+```bash
+coder providers create [NAME] \
+    --hostname=[HOSTNAME] \
+    --cluster-address=[CLUSTER_ADDRESS]
+```
+
+You must provide the following arguments:
+
+- `name`: A unique name of the workspace provider
+- `hostname`: The hostname of the workspace provider
+- `cluster-address`: The address of the Kubernetes cluster apiserver. This can be
```

Comment:
```suggestion
- `cluster-address`: The address of your Kubernetes cluster. You can retrieve this using:
```

Not sure the API Server is a useful distinction for those less familiar with Kubernetes, and people that do know the internals of Kubernetes to that degree will understand that the API server is the primary way we interact with it